### ISSUE TITLE
Error for 102 while training mnn model

### ISSUE THREAD

**Issue by ghost** [alibaba/MNN#925]
At the moment, I'm using this piece of code to re-train mnn model
```
class Quantizer:
    def __init__(self, cfg):
        self.cfg = cfg
        self.model = nn.load_module_from_file(cfg['COMMON']['model'], for_training=True)
        nn.compress.train_quant(self.model, quant_bits=8)
        self.opt = MNN.optim.SGD(1e-5, 0.9, 4e-5, MNN.optim.Regularization_Method.L2)
        self.opt.append(self.model.parameters)

    def quantize(self, loader):
        print("Start quantization...")
        self.model.train(True)
        for i in range(self.cfg['iter_steps']):
            print("Step: ", i + 1)
            example = loader.next()
            image = example[0][0]
            mask = [example[0][1], example[0][2]]
            target = example[1]
            output = self.model([image])
            loss = get_mnn_lost(target, output, mask)
            print("train loss: ", loss.read())
            self.opt.step(loss)

        self.model.train(False)
        pred = self.model.forward([F.placeholder([1, 368, 368, 3], F.NHWC)])
        F.save(pred, self.cfg['COMMON']['output'])
```
The loss is computed by: 
```
def get_mnn_lost(target, output, mask):
    return MNN.expr.abs(MNN.expr.reduce_mean((output[0] - target[0]) * mask[0])
                        + MNN.expr.reduce_mean((output[1] - target[1]) * mask[1]))
```

I observe the problem: if I use `opt.step(loss)`, I encounter `Error for 102`. If I don't, it runs okay and print loss after every iteration. Why does this happen?

**Comment by minhhoangbui** [comment]
@yyfcc17 Do you think in this case, we cannot use our custom loss function to give the optimizer, even though it remains `MNN.Var` type?

**Comment by yyfcc17** [comment]
of course we can use our own loss, i see no problem in your loss function now

**Comment by minhhoangbui** [comment]
@yyfcc17 Do you have any suggestion? When the optimizer isn't there, I can print the current loss using `loss.read()` without any error.
Anyway, I'd like to report a bug. If you set num_worker in DataLoader, the process doesn't work

**Comment by yyfcc17** [comment]
thank you, maybe these two are both known issue, we are fixing it. can you provide your mnn model file for us to debug?

**Comment by minhhoangbui** [comment]
@yyfcc17 You can find the model [here](https://drive.google.com/file/d/1pwn_EcsCeriaMgfgaw4xSZQwYP9pzmlF/view?usp=sharing) I use option `for-training` to create this model from tensorflow graph
